lib/instance.js

```diff
@@ -5,8 +5,10 @@ var Utils = require('./utils')
   , BelongsToMany = require('./associations/belongs-to-many')
   , InstanceValidator = require('./instance-validator')
   , QueryTypes = require('./query-types')
+  , Dottie = require('dottie')
   , Promise = require('./promise')
   , _ = require('lodash')
+  , primitives = ['string', 'number', 'boolean']
```

Comment:
I don't think it matters for dates. Unlses you compare on `valueOf` and i don't really see a reason to add the complexity.